src/Domain/Python/InertialCoordinates.cpp

```diff
@@ -0,0 +1,93 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Python/InertialCoordinates.hpp"
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <memory>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Utilities/GetOutput.hpp"
+
+namespace py = pybind11;
+
+namespace domain::py_bindings {
+
+namespace {
+template <size_t Dim>
+void bind_inertial_coordinates_impl(py::module& m) {  // NOLINT
+  m.def(
+      ("inertial_coordinates_" + get_output(Dim) + "d").c_str(),
+      [](const std::array<DataVector, Dim>& element_logical_coords,
+         const ElementId<Dim>& element_id, const Domain<Dim>& domain,
+         const std::optional<double> time = std::nullopt,
+         const std::optional<std::unordered_map<
+             std::string, const domain::FunctionsOfTime::FunctionOfTime&>>&
+             functions_of_time = std::nullopt) -> std::array<DataVector, Dim> {
+        // Transform std::array to Tensor
+        const tnsr::I<DataVector, Dim, Frame::ElementLogical> logical_coords(
+            element_logical_coords);
+        tnsr::I<DataVector, Dim, Frame::Inertial> inertial_coords{};
+
+        // Map logical to inertial coords
+        const auto& block = domain.blocks()[element_id.block_id()];
+        if (block.is_time_dependent()) {
+          if (not time.has_value()) {
+            throw std::runtime_error(
+                "The 'time' argument is required for time-dependent domains.");
```

Comment:
Do we have a way to test errors in python? The two `throw` statements don't seem to be tested.